user_config.c

```diff
@@ -1,94 +1,164 @@
 #include "user_config.h"
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #include "usb_app.h"
+#include "canlib/mcp2515/mcp_2515.h"
 
 typedef enum {
-	CHECK_CHAR,
-	CHECK_NUM,
-	CHECK_SEMI_COLON
+    CHECK_CHAR,
+    CHECK_NUM,
+    CHECK_SEMI_COLON,
+    CHECK_DATA
 } level;
 
 static level check_level = CHECK_CHAR;
 static uint8_t set_debug_level = 0;
 static uint8_t temp_num = 0;
 static bool set_sensor_messages = false;
+static bool sid_sent = false;
+static char data_to_send[6];
+static uint8_t data_index = 2;
+static uint8_t num_data_sent = 0;
+
 
 // save_type used to save what kind of message type is being checked ('G' for debug level, 'S' for sensor message, or 'L' for config)
 static char save_type;
 
 // function takes in a message and parses it to see if max_debug_level() or allow_sensor_messages() needs to be changed
-void parse_usb_string(const char *input) {
-	// Check for if the input is not NULL, if so: continue parsing the message
-	// Else: end function early
-	char config_msg[80];
-	char sensor_check[8];
-	uint8_t msg_length = 0;
-	if (input != NULL) {
-		for (uint8_t i = 0; input[i] != '\0'; i++) {
-			switch (check_level) {
-			// Case 2: last check for if there has been a valid message sent
-			// --> if parse_usb_string has received a semicolon
-			case CHECK_SEMI_COLON:
-				check_level = CHECK_CHAR;
-				if (input[i] == ';') {
-					switch (save_type) {
-					case 'G':
-						set_debug_level = temp_num;
-						break;
-					case 'S':
-						set_sensor_messages = temp_num;
-						break;
-					case 'L':
-						strcpy(sensor_check, "are");
-						if (!allow_sensor_messages())
-							strcat(sensor_check, " not");
-						msg_length = sprintf(config_msg, "Current Config: Max debug level = %d & Sensor messages %s allowed!\r\n", max_debug_level(), sensor_check);
-                        msg_length = strlen(config_msg);
-						usb_app_write_string(config_msg, msg_length);
-						break;
-					}
-				} // @suppress("No break at end of case")
+void parse_usb_string(const char *input)
+{
+    // Check for if the input is not NULL, if so: continue parsing the message
+    // Else: end function early
+
+    can_msg_t msg;
+    char config_msg[80];
+    char sensor_check[8];
+    char *nullptr;
+    data_to_send[0] = '0';
+    data_to_send[1] = 'x';
+    uint8_t msg_length = 0;
+    char char_to_check;
+    if (input != NULL) {
+        for (uint8_t i = 0; input[i] != '\0'; i++) {
+            if (input[i] >= 97)
```

Comment:
I see that what you're doing here is converting lowercase ascii letters to uppercase ones, but this is a non-portable and unsafe way of doing it (it will do bad things if you pass it a non-alphabetic character greater than 97, like `{`,`|`, or `}`. And it will handle delete character just like the letter `e`, which is bad.

The standard way of converting from lowercase to uppercase in C is to use the `toupper` standard library function, here: http://www.cplusplus.com/reference/cctype/toupper/.

This is my only comment on this code, so I'll make the slight modification and merge it now